Bali/src/main/java/OfferFilter.java

```diff
@@ -0,0 +1,25 @@
+import java.util.List;
+
+public class OfferFilter {
+    private String vendor;
```

Comment:
It's a good practice to add `final` to the fields that are set just once, in the constructor